spec.md

```diff
@@ -15,4 +16,574 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
 An implementation is not compliant if it fails to satisfy one or more of the MUST, REQUIRED, or SHALL requirements for the protocols it implements.
 An implementation is compliant if it satisfies all the MUST, REQUIRED, and SHALL requirements for the protocols it implements.
 
+## Terminology
+
+| Term              | Definition                                       |
+|-------------------|--------------------------------------------------|
+| Bucket            | A flat organization that contains immutable objects.                          |
+| Brownfield Bucket | An existing back-end bucket.                        |
+| Greenfield Bucket | A new back-end bucket created by COSI.                        |
+|COSI  System              | Container Orchestration system/ Container Object Storage Implementation, communicates with Plugins using COSI service RPCs.                                     |
+| SP                | Object Storage Provider, the vendor of a COSI plugin implementation.                                                          |
+| CO                | Container Orchestrator, like Kubernetes.                                                          |
+| RPC/gRPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                         |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a node ID. |
+| Driver(aka Plugin)           | Aka “plugin implementation”, a gRPC endpoint that implements the COSI Services.                                        |
+| Sidecar(aka Plugin Supervisor) | A COSI process that governs the lifecycle of a Plugin.                                                       |
+|COSI Node Adapter (aka CSI Driver) | A process that accepts gRPC calls from the Kubelet notifying it when the workload has been started and when it is terminating. |
+| Provisioner | A generic term meant to describe the combination of a sidecar and driver. "Provisioning" a bucket can mean creating a new bucket or granting access to an existing bucket.                                                       |
+| Workload          | The atomic unit of "work" scheduled by a COSI System. This MAY be a container or a collection of containers.                   |
+
+## Objective
+
+To define a standard “Container Object Storage Interface” (COSI) that will enable storage vendors (SP) to develop a plugin once and have it work across multiple Kubernetes Clusters and possibly on other COs.
```

Comment:
Why "possibly on other COs"?  Shouldn't the goal be to develop one plugin and have it work across a range of COs?

Reply:
Documenting the theoretical nature at the moment, I am not sure we have plans to test any other CO's.
Taking out 'possibly' to make it more affirmative as I do not see why it would not work.